packages/snap-controller/src/Recommendation/RecommendationController.ts

```diff
@@ -121,18 +137,19 @@ export class RecommendationController extends AbstractController {
 				this.eventManager.fire('track.product.impression', { controller: this, products: [result], trackEvent: data });
 				return data;
 			},
-			render: (result): RecommendationsSchemaData | undefined => {
-				if (this.events.product[result.id]?.render) return;
+			render: (result?: Product): RecommendationsSchemaData | undefined => {
+				const key = result?.id || 'noresults';
+				if (this.events.product[key]?.render) return;
 
-				const data = getRecommendationsSchemaData({ store: this.store, results: [result] });
+				const data = getRecommendationsSchemaData({ store: this.store, results: result ? [result] : [] });
 				this.tracker.events.recommendations.render({ data, siteId: this.config.globals?.siteId });
-				this.events.product[result.id] = this.events.product[result.id] || {};
-				this.events.product[result.id].render = true;
+				this.events.product[key] = this.events.product[key] || {};
+				this.events.product[key].render = true;
```

Comment:
Should probably revert, no results condition should not be handled via this method.